ironfish/src/primitives/target.ts

```diff
@@ -140,14 +141,31 @@ export class Target {
 
     // max(1 - (current_block_timestamp - parent_timestamp) / 10, -99)
     const diffInSeconds = (time.getTime() - previousBlockTimestamp.getTime()) / 1000
-    const sign = BigInt(Math.max(1 - Math.floor(diffInSeconds / 10), -99))
-    const offset = BigInt(previousBlockDifficulty) / BigInt(DIFFICULTY_ADJUSTMENT_DENOMINATOR)
 
     // diff = parent_diff + parent_diff / 2048 * max(1 - (current_block_timestamp - parent_timestamp) / 10, -99)
-    const difficulty = BigIntUtils.max(
-      BigInt(previousBlockDifficulty) + offset * sign,
-      Target.minDifficulty(),
-    )
+
+    const threshold = 1
+    const halfBucket = Math.floor(FREEZE_TIME_IN_SECONDS / 2)
+    const bucket =
+      Math.floor(
+        (diffInSeconds - TARGET_BLOCK_TIME_IN_SECONDS + halfBucket) / FREEZE_TIME_IN_SECONDS,
+      ) + 1
+
+    const max_change =
+      BigInt(previousBlockDifficulty) / BigInt(DIFFICULTY_ADJUSTMENT_DENOMINATOR)
+
+    let targetDifficulty = null
+    if (bucket <= threshold) {
+      // Scale up
+      const multiplier = threshold - bucket
+      targetDifficulty = previousBlockDifficulty + BigInt(max_change) * BigInt(multiplier)
+    } else {
+      // Scale down
+      const multiplier = Math.min(bucket - threshold, 99)
```

Comment:
When would this condition ever hit? it's basically saying choose the min of bucket from target, or 99. When would `bucket - threshold` ever be more than 99 for this condition to hit?  That would mean the block is trying to come in 1055 seconds from previous block, and we should set our difficulty to be the lowest difficulty possible which is basically checked on like 168 with `BigIntUtils.max(targetDifficulty, Target.minDifficulty())` ? 

I might be wrong, but I also don't see a test for this